### PR TITLE
https://issues.redhat.com/browse/ACM-7781 Update template_functions.adoc

### PR DESCRIPTION
**Pull request by dockerymick**
https://issues.redhat.com/browse/ACM-7781

### CODE REVIEW

File: governance/template_functions.adoc [hunk 1]
```diff
@@ -474,28 +474,41 @@ complianceType: musthave
...
+
+| String functions
+| `cat`, `contains`, `hasPrefix`, `hasSuffix`, `lower`, `quote`, `replace`, `split`, `splitn`, `substr`, `trim`, `trimAll`, `trunc`, `upper`
+
+| Version comparison
+| `semver`, `semverCompare`
+|===
+
+// `join`
+// `list` ---> assumed that list would be in the list library but it wasn't listed
```
Comment (mprahl) [inline governance/template_functions.adoc]: Yes, those should be in there. Good catch!

Here is the current list:
```go
	{
		"add",
		"append",
		"cat",
		"concat",
		"contains",
		"date",
		"default",
		"dig",
		"div",
		"empty",
		"fromJson",
		"has",
		"hasPrefix",
		"hasSuffix",
		"htpasswd",
		"join",
		"list",
		"lower",
		"mul",
		"mustAppend",
		"mustFromJson",
		"mustHas",
		"mustPrepend",
		"mustSlice",
		"mustToDate",
		"mustToRawJson",
		"now",
		"prepend",
		"quote",
		"replace",
		"round",
		"semver",
		"semverCompare",
		"slice",
		"split",
		"splitn",
		"sub",
		"substr",
		"ternary",
		"toDate",
		"toRawJson",
		"trim",
		"trimAll",
		"trunc",
		"until",
		"untilStep",
		"upper",
	}
```

Comment (dockerymick) [reply governance/template_functions.adoc]: @mprahl thanks for confirming, which library does `join` and `list` belong too? I checked the String library and List library, but those functions weren't present. Maybe i just missed them

File: governance/template_functions.adoc [hunk 2]
```diff
@@ -474,28 +474,41 @@ complianceType: musthave
...
+| Sprig library | Functions 
+
+| Cryptographic and security
+| `htpasswd`
+
+| Date
+| `date`, `mustToDate`, `now`, `toDate`
+
+| Default 
+| `default`, `empty`, `fromJson`, `mustFromJson`, `ternary`, `toJson`, `toRawJson`,
```
Comment (oafischer) [inline governance/template_functions.adoc]: Check for a superfluous comma at the end of each row, otherwise looks good!

**Review by oafischer — APPROVED** [review]
See one comment

File: governance/template_functions.adoc (hunk 1 above)
Comment (mprahl) [reply governance/template_functions.adoc]: @dockerymick `join` is documented here:
http://masterminds.github.io/sprig/string_slice.html

`lists` is the first one documented here:
http://masterminds.github.io/sprig/lists.html

Comment (dockerymick) [reply governance/template_functions.adoc]: Thanks for confirming!

File: governance/template_functions.adoc [hunk 3]
```diff
@@ -474,28 +474,41 @@ complianceType: musthave
...
+| `add`, `mul`, `div`, `round`, `sub` 
+
+| Integer slice
+| `until`, `untilStep`, 
+
+| Lists
+| `append`, `concat`, `has`, `mustAppend`, `mustHas`, `mustPrepend`, `mustSlice`, `prepend`, `slice` 
+
+| String functions
+| `cat`, `contains`, `hasPrefix`, `hasSuffix`, `lower`, `quote`, `replace`, `split`, `splitn`, `substr`, `trim`, `trimAll`, `trunc`, `upper`
```
Comment (dockerymick) [inline governance/template_functions.adoc]:
```suggestion
| `cat`, `contains`, `hasPrefix`, `hasSuffix`, `join`, `lower`, `quote`, `replace`, `split`, `splitn`, `substr`, `trim`, `trimAll`, `trunc`, `upper`
```

File: governance/template_functions.adoc [hunk 4]
```diff
@@ -474,28 +474,41 @@ complianceType: musthave
...
+| `dig` 
+
+| Integer math
+| `add`, `mul`, `div`, `round`, `sub` 
+
+| Integer slice
+| `until`, `untilStep`, 
+
+| Lists
+| `append`, `concat`, `has`, `mustAppend`, `mustHas`, `mustPrepend`, `mustSlice`, `prepend`, `slice`
```
Comment (dockerymick) [inline governance/template_functions.adoc]:
```suggestion
| `append`, `concat`, `has`, `list`, `mustAppend`, `mustHas`, `mustPrepend`, `mustSlice`, `prepend`, `slice` 
```

File: governance/template_functions.adoc [hunk 5]
```diff
@@ -474,28 +474,41 @@ complianceType: musthave
...
+
+| String functions
+| `cat`, `contains`, `hasPrefix`, `hasSuffix`, `lower`, `quote`, `replace`, `split`, `splitn`, `substr`, `trim`, `trimAll`, `trunc`, `upper`
+
+| Version comparison
+| `semver`, `semverCompare`
+|===
+
+// `join`
+// `list` ---> assumed that list would be in the list library but it wasn't listed
```
Comment (dockerymick) [inline governance/template_functions.adoc]:
```suggestion
```

File: governance/template_functions.adoc [hunk 6]
```diff
@@ -474,28 +474,41 @@ complianceType: musthave
...
+//Can you all help confirm the library for these functions? | MJ | 10/05/23
```
Comment (dockerymick) [inline governance/template_functions.adoc]:
```suggestion
```

**Comment by dockerymick** [comment]
@mprahl made updates according to your response. Is it important for the user to know about the newly supported sprig functions? Since it's community-based, it seems like it would not be needed on the What's new page

**Comment by mprahl** [comment]
> @mprahl made updates according to your response. Is it important for the user to know about the newly supported sprig functions? Since it's community-based, it seems like it would be needed on the What's new page

I think an entry saying that more Sprig functions are available in the "What's new" ACM page would be great!

**Review by mprahl — APPROVED** [review]

**Comment by dockerymick** [comment]
> > @mprahl made updates according to your response. Is it important for the user to know about the newly supported sprig functions? Since it's community-based, it seems like it would be needed on the What's new page
> 
> I think an entry saying that more Sprig functions are available in the "What's new" ACM page would be great!

Thanks Matt! I realize that I put would but meant wouldn't. Either way, thanks for confirming. I am adding the entry now and will merge

File: governance/template_functions.adoc [hunk 7]
```diff
@@ -474,28 +474,38 @@ complianceType: musthave
...
-* `semver`
-* `semverCompare`
-* `split`
-* `splitn`
-* `ternary`
-* `trim`
-* `until`
-* `untilStep`
-* `upper`
+Table of supported, community Sprig functions
```
Comment (dockerymick) [inline governance/template_functions.adoc]:
```suggestion
.Table of supported, community Sprig functions
```

**Pull request merged**